lib/googleauth/user_authorizer.rb

```diff
@@ -57,14 +58,19 @@ class UserAuthorizer
       # @param [String] callback_uri
       #  URL (either absolute or relative) of the auth callback.
       #  Defaults to '/oauth2callback'
-      def initialize client_id, scope, token_store, callback_uri = nil
+      # @param [String] code_verifier
+      #  Random string of 43-128 chars used to verify the key exchange using
+      #  PKCE. Auto-generated if not provided
+      def initialize client_id, scope, token_store,
+                     callback_uri = nil, code_verifier = nil
```

Comment:
I recommend using keyword arguments for additional optional arguments when the list gets this long. (I also recommend deprecating the existing `callback_uri` optional argument and providing a keyword argument replacement.)

Reply:
Updated `code_verifier` to be a keyword argument.

Reply:
I'll raise another PR for deprecating `callback_uri` .